libbeat/processors/add_cloud_metadata/provider_aws_ec2.go

```diff
@@ -18,125 +18,134 @@
+}
 
-	config := defaultConfig()
-	if err := c.Unpack(&config); err != nil {
-		logger.Warnf("error when load config for getting IMDSv2 token: %s. No token in the metadata request will be used.", err)
-		return ""
-	}
+var NewIMDSClient func(cfg awssdk.Config) IMDSClient = func(cfg awssdk.Config) IMDSClient {
+	return imds.NewFromConfig(cfg)
+}
 
-	tlsConfig, err := tlscommon.LoadTLSConfig(config.TLS)
-	if err != nil {
-		logger.Warnf("error when load TLS config for getting IMDSv2 token: %s. No token in the metadata request will be used.", err)
-		return ""
-	}
+type EC2Client interface {
+	DescribeTags(ctx context.Context, params *ec2.DescribeTagsInput, optFns ...func(*ec2.Options)) (*ec2.DescribeTagsOutput, error)
+}
 
-	client := http.Client{
-		Timeout: config.Timeout,
-		Transport: &http.Transport{
-			DisableKeepAlives: true,
-			DialContext: (&net.Dialer{
-				Timeout:   config.Timeout,
-				KeepAlive: 0,
-			}).DialContext,
-			TLSClientConfig: tlsConfig.ToConfig(),
-		},
-	}
+var NewEC2Client func(cfg awssdk.Config) EC2Client = func(cfg awssdk.Config) EC2Client {
+	return ec2.NewFromConfig(cfg)
+}
 
-	tokenReq, err := http.NewRequest("PUT", fmt.Sprintf("http://%s%s", metadataHost, ec2InstanceIMDSv2TokenURI), nil)
-	if err != nil {
-		logger.Warnf("error when make token request for getting IMDSv2 token: %s. No token in the metadata request will be used.", err)
-		return ""
-	}
+// AWS EC2 Metadata Service
+var ec2MetadataFetcher = provider{
+	Name: "aws-ec2",
 
-	tokenReq.Header.Add(ec2InstanceIMDSv2TokenTTLHeader, ec2InstanceIMDSv2TokenTTLValue)
-	rsp, err := client.Do(tokenReq)
-	if rsp == nil {
-		logger.Warnf("read token request for getting IMDSv2 token returns empty: %s. No token in the metadata request will be used.", err)
-		return ""
-	}
+	Local: true,
 
-	defer func(body io.ReadCloser) {
-		if body != nil {
-			body.Close()
+	Create: func(_ string, config *conf.C) (metadataFetcher, error) {
+		ec2Schema := func(m map[string]interface{}) mapstr.M {
+			m["service"] = mapstr.M{
+				"name": "EC2",
+			}
+			return mapstr.M{"cloud": m}
 		}
-	}(rsp.Body)
 
+		fetcher, err := newGenericMetadataFetcher(config, "aws", ec2Schema, fetchRawProviderMetadata)
+		return fetcher, err
+	},
+}
+
+// fetchRaw queries raw metadata from a hosting provider's metadata service.
+func fetchRawProviderMetadata(
+	ctx context.Context,
+	client http.Client,
+	result *result,
+) {
+	logger := logp.NewLogger("add_cloud_metadata")
+
+	// LoadDefaultConfig loads the EC2 role credentials
+	awsConfig, err := awscfg.LoadDefaultConfig(context.TODO(), awscfg.WithHTTPClient(&client))
 	if err != nil {
-		logger.Warnf("error when read token request for getting IMDSv2 token: %s. No token in the metadata request will be used.", err)
-		return ""
+		logger.Debugf("error loading AWS default configuration: %s.", err)
```

Comment:
This is an error message, is there a reason the log level is debug? Should be `Errorf`.

Reply:
`add_cloud_metadata` tries to fetch the metadata from all the available cloud providers, so if running on GCP - `fetchMetadata` for AWS will still run and in this case it might e confising to see an error on GCP regarding the AWS configuration. I've changed it to `Warnf` as it was done before - https://github.com/elastic/beats/pull/35182/commits/19ab15410b8407f67940ae1c79ada52060f93c9b